src/lightmap.cpp

```diff
@@ -571,44 +568,60 @@ void map::build_seen_cache()
             //
             // The naive solution of making the mirrors act like a second player
             // at an offset appears to give reasonable results though.
-            castLight( 1, 1.0f, 0.0f, 0, 1, 1, 0, mirror_pos.x, mirror_pos.y, offsetDistance );
-            castLight( 1, 1.0f, 0.0f, 1, 0, 0, 1, mirror_pos.x, mirror_pos.y, offsetDistance );
-
-            castLight( 1, 1.0f, 0.0f, 0, -1, 1, 0, mirror_pos.x, mirror_pos.y, offsetDistance );
-            castLight( 1, 1.0f, 0.0f, -1, 0, 0, 1, mirror_pos.x, mirror_pos.y, offsetDistance );
-
-            castLight( 1, 1.0f, 0.0f, 0, 1, -1, 0, mirror_pos.x, mirror_pos.y, offsetDistance );
-            castLight( 1, 1.0f, 0.0f, 1, 0, 0, -1, mirror_pos.x, mirror_pos.y, offsetDistance );
-
-            castLight( 1, 1.0f, 0.0f, 0, -1, -1, 0, mirror_pos.x, mirror_pos.y, offsetDistance );
-            castLight( 1, 1.0f, 0.0f, -1, 0, 0, -1, mirror_pos.x, mirror_pos.y, offsetDistance );
+            castLight( seen_cache, transparency_cache, 0, 1, 1, 0,
```

Comment:
As the `xx`, `xy`, `yx`, `yy` parameters are always compile time constants, have you considered moving them into template parameters of the function (e.f, like https://github.com/BevapDin/Cataclysm-DDA/commit/a9683a9ab6a8efb7cdd461c08be05581b127dffa)? That would avoid the need to copy when the function is called recursively, and it allows the compiler to optimize some calculations inside of it. As the variables always have the value 1, -1, or 0, and they are used like `deltaX * xx + deltaY * xy`, they can be optimized to simply `deltaX` or `deltaY` or `-deltaX`, not much computation left.